sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -55,11 +61,46 @@ public ClientLogger(Class<?> clazz) {
      * Retrieves a logger for the passed class name using the {@link LoggerFactory}.
      *
      * @param className Class name creating the logger.
-     * @throws RuntimeException it is an error.
+     * @throws RuntimeException in case of invalid arguments.
      */
     public ClientLogger(String className) {
+        this(className, Collections.emptyMap());
+    }
+
+    /**
+     * Retrieves a logger for the passed class using the {@link LoggerFactory}.
+     *
+     * @param clazz Class creating the logger.
+     * @param context Context to be populated on every log record written with this logger.
+     *                Objects are serialized with {@code toString()} method.
+     */
+    public ClientLogger(Class<?> clazz, Map<String, Object> context) {
+        this(clazz.getName(), context);
+    }
+
+    /**
+     * Retrieves a logger for the passed class name using the {@link LoggerFactory} with
+     * context that will be populated on all log records produced with this logger.
+     *
+     * <!-- src_embed com.azure.core.util.logging.clientlogger#globalcontext -->
+     * <pre>
+     * ClientLogger loggerWithContext = new ClientLogger&#40;ClientLoggerJavaDocCodeSnippets.class,
+     *     Map.of&#40;&quot;connectionId&quot;, &quot;95a47cf&quot;&#41;&#41;;
+     * loggerWithContext.info&#40;&quot;A formattable message. Hello, &#123;&#125;&quot;, name&#41;;
+     * </pre>
+     * <!-- end com.azure.core.util.logging.clientlogger#globalcontext -->
+     *
+     * @param className Class name creating the logger.
+     * @param context Context to be populated on every log record written with this logger.
+     *                Objects are serialized with {@code toString()} method.
+     * @throws RuntimeException in case of invalid arguments.
```

Comment:
Do we throw any other RuntimeException apart from `NullPointerException`? We should declare the specific exception types in the `@throws` clause.

Reply:
we are throwing something in `DefaultLogger` impl, but realistically we can't document closed set of exceptions here because they depend on SLF4J implementation used in the app.